Updated and cleaned up version of PR#1126 focusing on Evidence and Evaluation (#1161)
spdx/spdx-3-model: model/Core/Classes/EvaluationResult.md

SPDX-License-Identifier: Community-Spec-1.0

# EvaluationResult

## Summary

EvaluationResult is the result of an evaluation. 

## Description

EvaluationResult represents the documented outcome of assessing a work product, like a requirement, system, component, or process against specified criteria. It records whether the evaluated subject has met, failed, or yielded an inconclusive result based on the applied EvaluationResultType. The EvaluationResult captures the rationale behind the conclusion, providing necessary context and justification while referencing the specific RequirementVerification upon which the evaluation was based. This structure ensures clear traceability and supports compliance with standards for rigorous verification and validation processes.

> Note: An evaluation with a `inconclusive` EvaluationResultType should have a comment on it.

Reviewer: ```suggestion
An EvaluationResult with an evaluation value of "inconclusive" should have a value in its comment property.
```

- Make it more explicit that the comment is the comment of EvaluationResult
- I guess this is a requirement -- use "shall" for requirement / "should" for recommendation
  - See: https://github.com/spdx/spdx-3-model/blob/develop/docs/format.md#verbal-forms-for-expressions-of-provisions

Author: No, we discussed in the group that shall is not feasible for real usage. It is highly recommended to add a comment, but we cannot enforce it.

Reviewer: Thank you for explanation. Then using "should" is correct.
I have updated the suggestion.

## Metadata

- name: EvaluationResult
- SubclassOf: /Core/Element
- Instantiability: Concrete

## Properties

- evaluation
  - type: EvaluationResultType
  - minCount: 1
  - maxCount: 1

> - evalutationRationale

Reviewer 2: ```suggestion
- evaluationRationale
```
to match evaluationRationale.md

  - type: xsd:string
  - minCount: 1
  - maxCount: 1

> - evalutationBasedOn

Reviewer: ```suggestion
- evaluationBasedOn
```

The spec-parser failed because of a typo.
